src/objects/database.lzz

```diff
@@ -122,6 +140,126 @@ public:
 	}
 
 private:
+	class UpdateHook {
+	public:
+
+		explicit UpdateHook(
+			Addon * addon,
+			v8::Isolate * isolate,
+			Database * db,
+			v8::Local<v8::Function> fn
+		) :
+			addon(addon),
+			isolate(isolate),
+			db(db),
+			fn(isolate, fn) {}
+
+		static void invoke(void * data, int op, char const * dbName, char const * tableName, sqlite3_int64 rowid) {
+			UpdateHook * self = static_cast<UpdateHook *>(data);
+
+			v8::Isolate * isolate = self->isolate;
+			v8::Isolate::Scope isolateScope(isolate);
```

Comment:
Using `v8::Isolate::Scope` is new to this library, and I'm not familiar with it. What does it do? And do you think it should be used in other similar situations (such as user-defined SQL functions, or the `verbose` hook)?